dynacred/src/keypair.ts

```diff
@@ -0,0 +1,48 @@
+import { Log } from "@dedis/cothority";
+import { SignerEd25519 } from "@dedis/cothority/darc";
+import { curve, Point, Scalar } from "@dedis/kyber";
+const ed25519 = curve.newCurve("edwards25519");
+
+/**
+ * KeyPair implements some wrappers for the ed25519 private and public keys.
+ */
+export class KeyPair {
+
+    static rand(): KeyPair {
+        return KeyPair.fromPrivate(ed25519.scalar().pick());
+    }
+
+    static fromString(str: string): KeyPair {
+        const priv = ed25519.scalar();
+        priv.unmarshalBinary(Buffer.from(str, "hex"));
+        return KeyPair.fromPrivate(priv);
+    }
+
+    static fromPrivate(priv: Scalar | Buffer): KeyPair {
+        if (!isScalar(priv)) {
+            if (priv.length !== ed25519.scalarLen()) {
+                throw new Error("private key must be of length " + ed25519.scalarLen());
+            }
+            const privBuf = priv;
+            priv = ed25519.scalar();
+            priv.unmarshalBinary(privBuf);
+        }
+        const pub = ed25519.point().mul(priv);
+        return new KeyPair(priv, pub);
+    }
+
+    constructor(public priv: Scalar, public pub: Point) {
+    }
+
+    toString(): string {
+        return this.priv.marshalBinary().toString("hex");
+    }
+
+    signer(): SignerEd25519 {
+        return new SignerEd25519(this.pub, this.priv);
+    }
+}
+
+function isScalar(o: any): o is Scalar {
+    return "marshalBinary" in o;
```

Comment:
That looks like anything coming from kyber, isn't there something more precise?

Reply:
Changed it to `setBytes` and changed the signature of `isScalar`. 

What should be really done is to avoid uglifying `Scalar`, `Point`, `Buffer`, and others. Unfortunately I cannot find the reference in webpack anymore. All I can find is turning off _all_ mangling of class-names: https://github.com/webpack/webpack/issues/8132 - but I know I saw a post where it describes how to keep just _some_ classes complete.

Reply:
Ah, found it: https://webpack.js.org/configuration/externals/ - this would mean that @dedis/cothority should put @dedis/kyber as externals, and @c4dt/dynacred should probably do the same for @dedis/cothority and @dedis/kyber. Then you can import @dedis/{cothority,kyber} in the webapp, and be sure that you're using the same version, and that the classes still keep their name.